In binutils 2.35, 'nm -D' changed to show symbol versions along with
symbol names, with the usual @@ separator.  When generating
libtraceevent-dynamic-list we need just the names, so strip off the
version suffix if present.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
Tested-by: Salvatore Bonaccorso <carnil@debian.org>
Reviewed-by: Steven Rostedt <rostedt@goodmis.org>
Cc: linux-trace-devel@vger.kernel.org
Cc: stable@vger.kernel.org
Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>

diff -rupN linux-82c9666cb6fe999eb61f23c2c9d0d5dad7332fb6/tools/lib/traceevent/Makefile linux-patched/tools/lib/traceevent/Makefile
--- linux-82c9666cb6fe999eb61f23c2c9d0d5dad7332fb6/tools/lib/traceevent/Makefile	2020-07-20 10:31:30.000000000 +0200
+++ linux-patched/tools/lib/traceevent/Makefile	2021-08-04 16:08:49.946905559 +0200
@@ -250,9 +250,9 @@ define do_install_plugins
 endef
 
 define do_generate_dynamic_list_file
-	(echo '{';							\
-	$(NM) -u -D $1 | awk 'NF>1 {print "\t"$$2";"}' | sort -u;	\
-	echo '};';							\
+	(echo '{';									\
+	$(NM) -u -D $1 | awk 'NF>1 {sub("@.*", "", $$2); print "\t"$$2";"}' | sort -u;	\
+	echo '};';									\
 	) > $2
 endef
 
